### ISSUE TITLE
Parameter 'url' must be a string, not undefined is back v0.4.8

### ISSUE THREAD

**Issue by jitcoder** [Microsoft/vscode-chrome-debug#203]
Debugger console
`  ›******** Error in DebugAdapter - Unhandled promise rejection: Parameter 'url' must be a string, not undefined`

    {
      "name": "Attach to Chrome",
      "port": 9222,
      "type": "chrome",
      "request": "attach",
      "url": "http://shp-dev-03-pc:8080/",
      "webRoot": "${workspaceRoot}",
      "sourceMaps": true
    }

**Comment by jitcoder** [comment]
Sorry, it seems to have started to work again after restarting chrome.

**Comment by roblourens** [comment]
Would love to see the full logs from the `diagnosticLogging` option if it happens again.

**Comment by jitcoder** [comment]
will do if it happens again. Only other information I have is this happened right after the update and vscode restart.

**Comment by benlambt** [comment]
Is happening to me.

›OS: darwin x64
  ›Node: v5.10.0
  ›vscode-chrome-debug-core: 0.1.11
  ›debugger-for-chrome: 0.4.8
  ›Discovering targets via http://127.0.0.1:9222/json
  ›Error: Parameter 'url' must be a string, not undefined
    at Object.errP (/Users/Ben/.vscode/extensions/msjsdiag.debugger-for-chrome-0.4.8/node_modules/vscode-chrome-debug-core/out/src/utils.js:225:13)
    at /Users/Ben/.vscode/extensions/msjsdiag.debugger-for-chrome-0.4.8/node_modules/vscode-chrome-debug-core/out/src/chrome/chromeDebugAdapter.js:149:30
    at process._tickCallback (internal/process/next_tick.js:103:7)

**Comment by ben-x9** [comment]
{
  "version": "0.1.0",
  "configurations": [
    {
      "name": "Debug",
      "type": "chrome",
      "request": "attach",
      "port": 9222,
      "url": "http://localhost:8080",
      "sourceMaps": true,
      "webRoot": "${workspaceRoot}",
      "diagnosticLogging": true
    }
  ]
}

**Comment by ben-x9** [comment]
Works after restarting both vscode and chrome and removing "url": "http://localhost:8080".
However I originally got the error without the "url" option.

**Comment by ben-x9** [comment]
Not very useful though, as without the URL option it connects to the wrong tab

**Comment by ben-x9** [comment]
Everything seems to be working after restarting my system

**Comment by ben-x9** [comment]
I stopped the debugger and started it again, the error came back. Restarting VSCode doesn't help. The launch.json file is the same before and after.

**Comment by ben-x9** [comment]
OK, so the error occurs when I have chrome developer tools open. It runs fine as long as chrome dev tools is closed.

**Comment by roblourens** [comment]
Ah, I see the problem. I can't support using this and Chrome devtools at the same time, because Chrome only allows one devtools connection to the page. And apparently when something is already attached to the page, Chrome returns all the target info, but the websocket url is undefined. And I don't handle that correctly, and try to open a websocket on 'undefined'. Hence the error.

And I can imagine that if you have multiple tabs open, one of them with devtools attached, it could end up trying to attach to that one even if it's not the one you intended.

@jitcoder Were you also using chrome devtools?

**Comment by jitcoder** [comment]
in-fact I was!

**Comment by roblourens** [comment]
Cool, this will be fixed in the next update sometime next week (fixed in that it won't crash, but it still won't attach if chrome devtools is attached).

**Comment by jitcoder** [comment]
yeah it'd be nice if it messages the user "hey man I can't attach to this cuz you have devtools open"

**Comment by jitcoder** [comment]
in those exact words please. :smiley:

**Comment by baloodevil** [comment]
By the way, instead of getting the new friendly error, I just got the confusing error "Parameter 'url' must be a string, not undefined", even though I think the problem was that Chrome Dev Tools was attached.  Using the latest of pretty much everything. (Debugger for Chrome v2.5.2)

**Comment by roblourens** [comment]
Hm I don't see that, works for me.

**Comment by baloodevil** [comment]
I started a new ng-cli project and just tried getting breakpoints working.  Got that going.  Then couldn't figure out why a second project was not working (and giving this error).  I started by bumping up versions of several items in package.json.  That solved the problem.  The second project was not very old, and I did not take the time to figure out exactly which version dependency solved the issue.  But FYI, if anyone sees this again, do some maintenance on your package versions.  Thanks @roblourens for your work on this project.
